nemo_rl/utils/logger.py

```
@@ -1016,6 +1017,22 @@ def log_plot_token_mult_prob_error(
+        os.makedirs(os.path.dirname(filepath), exist_ok=True)
+
+        if filepath.endswith(".json"):
+            with open(filepath, "w") as f:
+                print(json.dumps(config, indent=2), end="", file=f)
+        elif filepath.endswith(".yaml"):
+            with open(filepath, "w") as f:
+                yaml.safe_dump(config, f)
+        else:
+            raise NotImplementedError
+
+        print(f"Logged config dict to {repr(filepath)}", flush=True)
```

Ruff (0.13.1)

Use explicit conversion flag

Replace with conversion flag

(RUF010)